### ISSUE TITLE
Support plugins in sigs/scheduler-plugins?

### ISSUE THREAD

**Issue by sanposhiho** [kubernetes-sigs/kube-scheduler-simulator#77]
https://github.com/kubernetes-sigs/scheduler-plugins

This idea should be discussed if it is really needed or not.

If we want to support it, we also need to support CRD (like [podgroup](https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/pkg/coscheduling/README.md#podgroup)). Supporting each CRDs can be a big task. Therefore, we need to think of some better way to do that, like handle CRDs together... etc (I have no good vision for it now 😓  )

But, #35 will make it easy to use custom-scheduler. So, even if they are not supported here, users will be able to use that plugins with #35 feature. In that case, since CRDs are not supported in the WebUI, users need to create CRD from another client. #55 will enable other clients (like kubectl, client-go... etc) to communicate with api-server.

To sum up my opinion, if there are many users who want to use plugins in sigs/scheduler-plugins, it is better to support them and support CRD in webui as well. But if not, we don't need to support them. 

And there is a possibility that "scheduler-plugins will not be supported but a handling CRDs feature will be.".

**Comment by matthewygf** [comment]
Hi @sanposhiho , I have an experimental [branch](https://github.com/matthewygf/kube-scheduler-simulator/tree/crds) that made CRDs work , well at least it accepts the CRD and create them appropriately.

The idea is copying how kube-api-server create its [aggregate server](https://github.com/kubernetes/kubernetes/blob/30a21e9abdbbeb78d2b7ce59a79e46299ced2742/cmd/kube-apiserver/app/server.go#L169) and extending it into the simulator - k8sapiserver.

It is quite a huge change to the simulator, so I am not sure whether you would like this added...

**Comment by marcobarlo** [comment]
I would be interested in simulating my custom plugins too for testing reasons on scale, is that possible in your branch @matthewygf?

**Comment by matthewygf** [comment]
@marcobarlo , it depends whether your custom plugins implement the "Filter/Score" interface. Also It depends whether you need to handle any CRDs, in which case you might need to add a controller instance within the simulator. 

If you don't need to handle any CRDs, you can probably fork and register your plugin in the simulator (probably need to add a few lines of code).

**Comment by sanposhiho** [comment]
I think we can close this by #121.
Thanks a lot @matthewygf !

/close
